fdb-record-layer-core/src/test/java/com/apple/foundationdb/record/provider/foundationdb/OnlineIndexerUniqueIndexTest.java

```diff
@@ -366,4 +371,184 @@ public void resolveUniquenessViolations() {
             context.commit();
         }
     }
+
+    @Test
+    public void testReadableUniquePendingSingle() {
+        final String indexName = "simple$value_2";
+        List<TestRecords1Proto.MySimpleRecord> records = LongStream.range(0, 10).mapToObj( val ->
+                TestRecords1Proto.MySimpleRecord.newBuilder().setRecNo(val).setNumValue2(((int)val) % 5).build()
+        ).collect(Collectors.toList());
+        Index index = new Index(indexName, field("num_value_2"), EmptyKeyExpression.EMPTY, IndexTypes.VALUE, IndexOptions.UNIQUE_OPTIONS);
+        FDBRecordStoreTestBase.RecordMetaDataHook hook = metaDataBuilder -> metaDataBuilder.addIndex("MySimpleRecord", index);
+
+        openSimpleMetaData();
+        try (FDBRecordContext context = openContext())  {
+            records.forEach(recordStore::saveRecord);
+            context.commit();
+        }
+        openSimpleMetaData(hook);
+        try (FDBRecordContext context = openContext()) {
+            recordStore.markIndexWriteOnly(index).join();
+            context.commit();
+        }
+        openSimpleMetaData(hook);
+        try (OnlineIndexer indexBuilder = OnlineIndexer.newBuilder()
+                .setDatabase(fdb).setMetaData(metaData).setIndex(index).setSubspace(subspace)
+                .setIndexingPolicy(OnlineIndexer.IndexingPolicy.newBuilder()
+                        .allowUniquePendingState())
+                .build()) {
+            indexBuilder.buildIndex(true);
+        }
+        try (FDBRecordContext context = openContext()) {
+            final IndexState indexState = recordStore.getIndexState(indexName);
+            assertEquals(IndexState.READABLE_UNIQUE_PENDING, indexState);
+            assertTrue(recordStore.isIndexReadableUniquePending(indexName));
+            context.commit();
+        }
+        // now try resolving it, and marking readable with another build
```

Comment:
In all of these tests, it would probably be good to scan the index, and assert about the results.

Reply:
Added to some.

Reply:
I see a lot of code that calls `scanUniquenessViolations`, but none that actually scan the values of the index and assert that it is scannable and has the right values.

Reply:
True. But can it be done in another PR? It doesn't seem to be directly related to this new state. 
(For the record, we might also want to change the usage of `scanUniquenessViolations` to something like: 
```
try(RecordCursor<RecordIndexUniquenessViolation> cursor = recordStore.scanUniquenessViolations(index) {
...
}
```